dali/test/python/test_operator_numpy_reader.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import nvidia.dali as dali
+import numpy as np
+from numpy.testing import assert_array_equal, assert_allclose
+import os
+import tempfile
+
+class NumpyReaderPipeline(Pipeline):
+    def __init__(self, path, batch_size, path_filter="*.npy", num_threads=1, device_id=0, num_gpus=1):
+        super(NumpyReaderPipeline, self).__init__(batch_size,
+                                                  num_threads,
+                                                  device_id)
+        
+        self.input = ops.NumpyReader(file_root = path,
+                                     file_filter = path_filter,
+                                     shard_id = device_id,
+                                     num_shards = num_gpus)
+
+        
+    def define_graph(self):
+        inputs = self.input(name="Reader")
+
+        return inputs
+
+# test: compare reader with numpy
+# with different batch_size and num_threads
+
+def test_reader_vs_numpy():
+
+    # create temporary directory
+    with tempfile.TemporaryDirectory() as test_data_root:
+    
+        # we need that
+        rng = np.random.RandomState(12345)
+    
+        # 1D array integer
+        arr = rng.randint(0,10, (15))
+        np.save(os.path.join(test_data_root, "test_01.npy"), arr)
+
+        # 1D array float
+        arr = rng.random_sample((25)).astype(np.float32)
```

Comment:
How about testing more data types that numpy supports?

Reply:
fixed